bundles/org.openhab.binding.bluetooth.daikinmadoka/src/main/java/org/openhab/binding/bluetooth/daikinmadoka/internal/model/commands/BRC1HCommand.java

```diff
@@ -0,0 +1,111 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.bluetooth.daikinmadoka.internal.model.commands;
+
+import java.util.concurrent.Executor;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.locks.Condition;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+
+/**
+ *
+ * @author blafois
+ *
+ */
+@NonNullByDefault
+public abstract class BRC1HCommand {
+
+    public enum State {
+        NEW,
+        ENQUEUED,
+        SENT,
+        SUCCEEDED,
+        FAILED
+    }
+
+    private volatile @Nullable State state;
+
+    private final Lock stateLock = new ReentrantLock();
+
+    private final Condition stateCondition = stateLock.newCondition();
+
+    public abstract boolean handleResponse(Executor executor, ResponseListener listener, byte @Nullable [] response);
```

Comment:
To reduce the amount of redundant logic that subclasses must implement I suggest changing this method signature so that all the redundant logic is handled in by the method caller instead:
```suggestion
    public abstract void handleResponse(ResponseListener listener, MadokaMessage mm);
```